api/v2/api.go

```diff
@@ -508,7 +508,50 @@ func (api *API) getSilencesHandler(params silence_ops.GetSilencesParams) middlew
 		sils = append(sils, &silence)
 	}
 
-	return silence_ops.NewGetSilencesOK().WithPayload(sils)
+	active := open_api_models.GettableSilences{}
+	pending := open_api_models.GettableSilences{}
+	expired := open_api_models.GettableSilences{}
+
+	for _, s := range sils {
+		switch *s.Status.State {
+		case "active":
+			active = append(active, s)
+		case "pending":
+			pending = append(pending, s)
+		case "expired":
+			expired = append(expired, s)
+		}
+	}
+
+	sort.Slice(active, func(i int, j int) bool {
+		endsAt1 := apiTimeToGoTime(active[i].Silence.EndsAt)
```

Comment:
It would be more efficient to convert `*strfmt.DateTime` back to `time.Time` directly:

```go
endsAt1 := time.Time(*active[i].Silence.EndsAt)
```

Reply:
Yes, that's much better, thanks!